sdks/python/apache_beam/transforms/combiners.py

```diff
@@ -193,9 +193,13 @@ def Of(pcoll, n, compare=None, *args, **kwargs):
           original_compare = compare
           compare = lambda a, b: original_compare(b, a)
       # This is a more efficient global algorithm.
+      top_per_bundle = pcoll | core.ParDo(_TopPerBundle(n, compare))
+      # If pcoll is empty, we can't guerentee that top_per_bundle
+      # won't be empty, so inject at least one empty accumulator
+      # so that downstream is guerenteed to produce non-empty output.
+      empty_bundle = pcoll.pipeline | core.Create([(None, [])])
```

Comment:
Out of curiosity, how do the coders for this work out since Flatten requires a coder that can encode top_per_bundle and empty_bundle?

Reply:
Runners are responsible for resolving this based on type hints, e.g. by creating a coder that can handle the union type. (In practice here, whatever can encode the LHS can also encode the RHS.)